### PR TITLE
Fixes for mid-line comments and multiline text blocks in mmCIF files

### PR DESCRIPTION
**Pull request by StefansM**
This pull request addresses issue #1441.

The changes to lines 36&ndash;41 of `test_PDB_MMCIF2Dict.py` were necessary to correct the test. The field `_struct_ref.pdbx_seq_one_letter_code` should include the newlines, but the test previously accepted the concatenated result as correct.

The test could do with some cleanup (line 35 is over 6,000 characters long), but that doesn't fall within the scope of this pull request.

----
I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ AND the _BSD 3-Clause License_.

I am happy to be thanked by name in the ``NEWS.rst`` and
``CONTRIB.rst`` files.

I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

### CODE REVIEW

File: Tests/test_PDB_MMCIF2Dict.py [hunk 1]
```diff
@@ -21,6 +21,9 @@
 
 from Bio.PDB.MMCIF2Dict import MMCIF2Dict
 
+import io
```
Comment (peterjc) [inline Tests/test_PDB_MMCIF2Dict.py]: Using ``io.StringIO`` makes Python 2 and 3 at once tricky, which would explain why you needed the explicit unicode strings later.

Why we normally use ``from Bio._py3k import StringIO`` which does ``from StringIO import StringIO`` under Python 2 or ``from io import StringIO`` under Python 3.

Either was seems fine (and the difference only matters under Python 2 anyway which will eventually go away).

**Comment by peterjc** [comment]
I'd like @jgreener64 to take a look at the code, but from a technical perspective it looks nicely done with tests. Good job.

Would you like to add yourself to the ``NEWS.rst`` and ``CONTRIB.rst`` files? That lets you decide exactly how to list your name (which you do not currently openly disclose on your GitHub profile).

Note the names are sorted alphabetically (see comment at top of the contributors file, this is checked by TravisCI). Alternatively I can do that by adding a commit to your branch.

File: Tests/test_PDB_MMCIF2Dict.py (hunk 1 above)
Comment (StefansM) [reply Tests/test_PDB_MMCIF2Dict.py]: Ah-ha! I thought there might be something like that available, but missed `Bio._py3k`. I don't develop on Python 2, but the build bot did its job nicely and let me know there was a problem. I don't mind switching to `Bio._py3k` if you consider that more idiomatic for biopython.

Comment (peterjc) [reply Tests/test_PDB_MMCIF2Dict.py]: Sure - my guess is you can use ``from Bio._py3k import StringIO`` and remove the ``u`` prefixes.

Note that in general I would recommend using the third-party library ``six`` for this kind of compatibility shim, but we wanted to keep the Biopython dependencies to a minimum and have a few other more project specific hacks under ``Bio._py3k`` anyway.

**Comment by peterjc** [comment]
Ah - I missed you were already in the ``CONTRIB.rst`` file. Thanks for your further contribution 👍

**Comment by StefansM** [comment]
Thanks Peter. I'm already listed correctly in `CONTRIB.rst`, so I've just added myself to `NEWS.rst`.

**Comment by jgreener64** [comment]
Having looked at the code I am happy with it, nice one @StefansM.

**Comment by peterjc** [comment]
Thank you both :)

**Pull request merged**